### DISCUSSION TITLE
Capacitor v3: <Plugin> is not implemented on android

### DISCUSSION THREAD

**Discussion by antonio-cabezas** [ionic-team/capacitor#4918] [Q&A]
I am using Capacitor v3 with a React PWA and I have the following problem: 

When I build my app for Android it works fine but if the app reaches the OnDestroy lifecycle event and I execute it again I can see this lines printed on Logcat:

> Capacitor: Unable to read file at path public/plugins

> Capacitor/Console: File: https://example.com/static/js/8.cac4fa54.chunk.js - Line 2 - Msg: Uncaught (in promise) Error: "Storage" plugin is not implemented on android

Last line is duplicated for all Capacitor official plugins that I am using.

I have the following dependencies:

    "@capacitor-community/facebook-login": "3.1.1",
    "@capacitor/android": "3.1.2",
    "@capacitor/app": "^1.0.2",
    "@capacitor/core": "^3.1.2",
    "@capacitor/device": "^1.0.2",
    "@capacitor/geolocation": "^1.0.2",
    "@capacitor/ios": "3.1.2",
    "@capacitor/network": "^1.0.2",
    "@capacitor/push-notifications": "^1.0.3",
    "@capacitor/status-bar": "^1.0.2",
    "@capacitor/storage": "^1.0.3",
    "@codetrix-studio/capacitor-google-auth": "3.0.2",

And also as dev dependency:

    "@capacitor/cli": "^3.1.2",

My MainActivity (here I registered community plugins):

    import android.os.Bundle;
    
    import com.codetrixstudio.capacitor.GoogleAuth.GoogleAuth;
    import com.getcapacitor.BridgeActivity;
    import com.getcapacitor.community.facebooklogin.FacebookLogin;
    
    public class MainActivity extends BridgeActivity {
        @Override
        public void onCreate(Bundle savedInstanceState) {
            super.onCreate(savedInstanceState);
            registerPlugin(GoogleAuth.class);
            registerPlugin(FacebookLogin.class);
        }
    }

The capacitor config is the following:

    {
      "appId": "com.example.frontend",
      "appName": "Example",
      "bundledWebRuntime": false,
      "webDir": "build",
      "npmClient": "yarn",
      "plugins": {
        "GoogleAuth": {
          "scopes": ["profile", "email"],
          "serverClientId": "example-server-id.apps.googleusercontent.com",
          "forceCodeForRefreshToken": true
        }
      },
      "server": {
        "allowNavigation": ["example.com", "*.example.com"],
        "hostname": "example.com",
        "androidScheme": "https"
      },
      "cordova": {}
    }

Any suggestions are appreciated.

**Comment by antonio-cabezas** [comment]
It happens only if I register the service worker. 

Since it's not necessary to have the service worker registered on the native app, I am registering it depending on the device platform.

**Reply by jcesarmobile** [reply]
You can do this to make service workers work on android SDK 24+ devices, but yeah, it’s better if you don’t put them there since the app files are already bundled with the app.

https://github.com/ionic-team/capacitor/issues/1655#issuecomment-579229390

**Comment by Elvincth** [comment]
Also I find out that if you are in live reload mode, cap3 plugins will no wotk even you synced it into the andorid project.
